documentation/Readme.md

```diff
@@ -0,0 +1,47 @@
+# Project Title
+- Cool logo
+
+---
+
+### Introduction
+
+- Summary of the purpose and function of this project
+- Provide links to Issue queue and project page (if any)
+
+### Requirements
+
+- Hardware requirements (Processor/Storage/RAM/GPU)
+- OS requirements (List of compatible Operating Systems)
+- List of Dependencies
+
+### Installation
+
+- Guide/code to install/build dependencies for respective OS
+- If a dependency is to be built, provide a link to its *Installation* page. Do not reinvent the wheel here.
+- Instructions to build the project after the dependencies have been met
+
+### Configuration
+
+- List any configuration is required; for example, the various config options in cmake.
+- Any special notes about the configuration of this module – including but not limited to permissions.
+
+### Testing
+
+- Instructions to run unit tests.
+- Each unit test should have internal documentation on what the test is checking for, and the logic that is being used.
+
+- Instructions for Integration Testing
+
+- Instructions to follow when tests fail. (This may also be included int eh Troubleshooting section)
+
+### Troubleshooting and FAQ
```

Comment:
Is it a good idea to have FAQ in the README file? The list could be very big.
How about creating a github wiki page and hosting the FAQ there. Here we can just give the link to the FAQ page here.

Reply:
Good idea. I think we can write 3-5 of the *most* frequently asked questions here, only those which we think can save the user's time; and then provide a link to Wiki/Project Page/Issue Queue.

Reply:
yeah